CMakeLists.txt

```diff
@@ -359,6 +359,7 @@ SET(PROD_AVX2_MICROKERNEL_SRCS src/amalgam/gen/avx2.c)
 SET(PROD_AVX512F_MICROKERNEL_SRCS src/amalgam/gen/avx512f.c)
 SET(PROD_AVX512SKX_MICROKERNEL_SRCS src/amalgam/gen/avx512skx.c)
 SET(PROD_AVX512VBMI_MICROKERNEL_SRCS src/amalgam/gen/avx512vbmi.c)
+SET(PROD_AVX512VNNI_MICROKERNEL_SRCS src/amalgam/gen/avx512vnni.c)
```

Comment:
The build fails because src/amalgam/gen/avx512vnni.c is not found, such as https://github.com/google/XNNPACK/actions/runs/6175860668/job/16763598615?pr=5467

Should it be generated at https://github.com/google/XNNPACK/blob/test_565082406/scripts/generate-amalgamation.sh#L27?

For example:
```
tools/amalgamate-microkernels.py -i immintrin.h -s PROD_AVX512VNNI_MICROKERNEL_SRCS -o src/amalgam/gen/avx512vnni.c
```